### PR TITLE
Add buildscript for MacOSX and iOS

### PR DESCRIPTION
**Pull request by Ghabry**
Requires MacOSX, no osxcross support

This PR is mostly for discussion. The build_easyrpg.sh is just a reference for Jenkins, will remove it later.

The toolchains are already placed where the Jenkins job would put them ($JENKINS/ios-toolchain/ios and $JENKINS/osx-toolchain/osx) for playing around.

### CODE REVIEW

File: osx/2_build_toolchain.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	cd libpng-1.6.24
+	perl -pi -e 's/^bin_PROGRAMS/# $&/' Makefile.am
+	autoreconf -fi
+	cd ..
+
+        # disable SDL2 mixer examples
+	cd SDL2_mixer-2.0.1
+	echo 'AC_CONFIG_MACRO_DIRS([acinclude])' | cat - configure.in > configure.tmp.in && mv configure.tmp.in configure.in
+	perl -pi -e 's/ \$\(objects\)\/playwave\$\(EXE\) \$\(objects\)\/playmus\$\(EXE\)//' Makefile.in
+	autoreconf -fi
```
Comment (carstene1ns) [inline osx/2_build_toolchain.sh]: might be easier using a patch here.

File: ios/1_download_library.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+function git_clone {
+	url=$1
+	file=${url##*/}
+	msg "Cloning $file..."
+
+	git clone --depth=1 $url
+}
+
+#msg " [1] Installing devkitPPC"
```
Comment (carstene1ns) [inline ios/1_download_library.sh]: leftover.

**Comment by Ghabry** [comment]
Compiles for me now...

**Comment by Ghabry** [comment]
That toolchain+jenkins should be correct now, got liblcf+Player to compile.

File: ios/3_cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+echo
+echo "Cleaning up library build folders and other stuff..."
+
+rm -rf zlib-*/ freetype-*/ harfbuzz-*/ icu/ icu-native/ libpng-*/ expat-*/ pixman-*/ mpg123-*/ \
+	libsndfile-*/ libxmp-lite-*/ speexdsp-*/ wildmidi-*/ libogg-*/ libvorbis-*/ \
+	SDL2_mixer-*/ SDL2-*/
+rm -f *.bz2 *.gz *.xz *.tgz *.pl icudt* .patches-applied
```
Comment (carstene1ns) [inline ios/3_cleanup.sh]: No `SDL2_mixer` and no `*.pl` needed.

**Pull request merged**